x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -5,11 +5,50 @@
  * 2.0.
  */
 import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
-import { ApmDataSource } from '../../../common/data_source';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 import { ApmDocumentType } from '../../../common/document_type';
 import { RollupInterval } from '../../../common/rollup';
 import { APMEventClient } from './create_es_client/create_apm_event_client';
 import { getConfigForDocumentType } from './create_es_client/document_type';
+import { TRANSACTION_DURATION_SUMMARY } from '../../../common/es_fields/apm';
+import { TimeRangeMetadata } from '../../../common/time_range_metadata';
+
+const getFilters = ({
+  documentType,
+  rollupInterval,
+  filters,
+}: {
+  documentType: ApmDocumentType;
+  rollupInterval: RollupInterval;
+  filters: estypes.QueryDslQueryContainer[];
+}) => {
+  const searchParams = {
+    apm: {
+      sources: [
+        {
+          documentType,
+          rollupInterval,
+        },
+      ],
+    },
+    body: {
+      track_total_hits: 1,
+      size: 0,
+      terminate_after: 1,
+    },
+  };
+  return {
+    ...searchParams,
+    body: {
+      ...searchParams.body,
+      query: {
+        bool: {
+          filter: filters,
+        },
+      },
+    },
+  };
+};
 
 export async function getDocumentSources({
```

Comment:
Can you add a unit test for `getDocumentSources `?

Reply:
You are right. We should cover this with API tests, and we already have `time_range_metadata.spec.ts`.

Reply:
Yes, i extended and added more API tests to `time_range_metadata.spec.ts` to cover the newly added scenarios